Wrong comment and unsatisfied trait bound.

### Describe the bug

1. In zenoh-main/commons/zenoh-protocol/src/core/encoding.rs, the trait bound `ZSlice: Hash` is not satisfied.
2. In zenoh-main/io/zenoh-transport/src/common/seq_num.rs, following 2 functions' comments are wrong.
First, this function will return Error instead of panic if `value` or `initial_sn` is greater or equal than `resolution`. So the comment should be "This function will return error if ..." instead of "This function will panic if ...".
Second, for the second function, comment should use 'initial_sn' instead of `value`. The second comment should be "This function will return error if `initial_sn` is out of bound w.r.t. `resolution`. That is if `initial_sn` is greater or equal than `resolution`."
```rust
/// This function will panic if `value` is out of bound w.r.t. `resolution`. That is if
    /// `value` is greater or equal than `resolution`.
    ///
    pub(crate) fn make(value: TransportSn, resolution: Bits) -> ZResult<SeqNum> {
        let mask = get_mask(resolution);
        let mut sn = SeqNum { value: 0, mask };
        sn.set(value)?;
        Ok(sn)
    }
/// This function will panic if `value` is out of bound w.r.t. `resolution`. That is if
    ///   `value` is greater or equal than `resolution`.
    ///
    pub(crate) fn make(initial_sn: TransportSn, resolution: Bits) -> ZResult<SeqNumGenerator> {
        let sn = SeqNum::make(initial_sn, resolution)?;
        Ok(SeqNumGenerator(sn))
    }
```

### To reproduce

For problem 1: In zenoh-main/io/zenoh-transport, run `cargo test` and following error will emerged.
```rust
error[E0277]: the trait bound `ZSlice: Hash` is not satisfied
  --> commons/zenoh-protocol/src/core/encoding.rs:32:5
   |
29 | #[cfg_attr(feature = "std", derive(Hash))]
   |                                    ---- in this derive macro expansion
...
32 |     pub schema: Option<ZSlice>,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Hash` is not implemented for `ZSlice`
   |
   = help: the trait `Hash` is implemented for `std::option::Option<T>`
   = note: required for `std::option::Option<ZSlice>` to implement `Hash`
   = note: this error originates in the derive macro `Hash` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `zenoh-protocol` (lib) due to 1 previous error
```
For problem 2: In zenoh-main/io/zenoh-transport/src/common/seq_num.rs, following test will be failed.
```rust
#[test]
#[should_panic]
fn test(){
    let _ = SeqNum::make(u8::MAX as TransportSn, Bits::U8);
}
```

### System info

- Platform: MacOS

--- a/io/zenoh-transport/src/common/seq_num.rs
+++ b/io/zenoh-transport/src/common/seq_num.rs
@@ -57,7 +57,7 @@ impl SeqNum {
     ///   - 16_386 (i.e., 2^14)
     ///   - 2_097_152 (i.e., 2^21)
     ///
-    /// This function will panic if `value` is out of bound w.r.t. `resolution`. That is if
+    /// This function will return an error if `value` is out of bound w.r.t. `resolution`. That is if
     /// `value` is greater or equal than `resolution`.
     ///
     pub(crate) fn make(value: TransportSn, resolution: Bits) -> ZResult<SeqNum> {
@@ -179,8 +179,8 @@ impl SeqNumGenerator {
     ///   As a consequence of wire zenoh's representation of sequence numbers
     ///   this should be a multiple of 7.
     ///
-    /// This function will panic if `value` is out of bound w.r.t. `resolution`. That is if
-    ///   `value` is greater or equal than `resolution`.
+    /// This function will return an error if `initial_sn` is out of bound w.r.t. `resolution`. That is if
+    ///   `initial_sn` is greater or equal than `resolution`.
     ///
     pub(crate) fn make(initial_sn: TransportSn, resolution: Bits) -> ZResult<SeqNumGenerator> {
         let sn = SeqNum::make(initial_sn, resolution)?;

--- a/io/zenoh-transport/src/common/seq_num.rs
+++ b/io/zenoh-transport/src/common/seq_num.rs
@@ -57,6 +57,8 @@ impl SeqNum {
     ///   - 16_386 (i.e., 2^14)
     ///   - 2_097_152 (i.e., 2^21)
     ///
+    /// # Errors
+    ///
     /// This function will return an error if `value` is out of bound w.r.t. `resolution`. That is if
     /// `value` is greater or equal than `resolution`.
     ///
@@ -178,6 +180,8 @@ impl SeqNumGenerator {
     /// * `sn_resolution` - The resolution, in bits, to be used for the sequence number generator.
     ///   As a consequence of wire zenoh's representation of sequence numbers
     ///   this should be a multiple of 7.
+    ///
+    /// # Errors
     ///
     /// This function will return an error if `initial_sn` is out of bound w.r.t. `resolution`. That is if
     ///   `initial_sn` is greater or equal than `resolution`.

